### PR TITLE
Fix false positive ``super-init-not-called`` for inherited ``init``

### LINKED ISSUE #4941: False positive for ``super-init-not-called``
### Bug description

```python
class A:
    def __init__(self):
        pass


class B(A):
    pass


class D(B, A):
    def __init__(self):
        A.__init__(self)
```

This was found while working on #4940, after fixing this we might want to check if there are any messages that are unnecessarily disabled in our own codebase.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
pylint test.py
************* Module test
test.py:11:4: W0231: __init__ method from base class 'B' is not called (super-init-not-called)

------------------------------------------------------------------
Your code has been rated at 8.75/10 (previous run: 8.75/10, +0.00)
```

### Expected behavior

As class `B` does not have a `__init__` it can't be called. The only `__init__` included in class `B` is the `__init__` of class `A` which is being called in class `C`.

### Pylint version

```shell
pylint 2.10.3-dev0
astroid 2.7.3
Python 3.9.6 (default, Aug  2 2021, 08:23:35) 
[Clang 12.0.5 (clang-1205.0.22.11)]
```

### OS / Environment

macOS

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by DanielNoord**
- [x] Add yourself to CONTRIBUTORS if you are a new contributor.
- [x] Add a ChangeLog entry describing what your PR does.
- [x] If it's a new feature, or an important bug fix, add a What's New entry in
      `doc/whatsnew/<current release.rst>`.
- [x] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #4941

### CODE REVIEW

File: pylint/checkers/imports.py [hunk 1]
```diff
@@ -427,9 +427,7 @@ class ImportsChecker(DeprecatedMixin, BaseChecker):
         ),
     )
 
-    def __init__(
-        self, linter: Optional["PyLinter"] = None
-    ):  # pylint: disable=super-init-not-called # See https://github.com/PyCQA/pylint/issues/4941
+    def __init__(self, linter: Optional["PyLinter"] = None) -> None:
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/imports.py:430]: This diff is satisfying ! 🙏

**Comment by DanielNoord** [comment]
What's up with coveralls again?... I'm seriously debating checking out if codecov is better for us 😓

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM, it's probably going to need a rebase after we merge #5697, or we could merge first but the comment in 5697 would be obsolete.

**Comment by DanielNoord** [comment]
Let's do the other one first! I did not touch the comments and docstrings in this one as I imagined the other one would be merged first!

**Comment by DanielNoord** [comment]
Merge conflicts have been resolved. This should pass tests :)

**Review by Pierre-Sassoulas — APPROVED** [review]
👍

**Pull request merged**